app/validators/array_length_validator.rb

```diff
@@ -0,0 +1,9 @@
+class ArrayLengthValidator < ActiveModel::EachValidator
+
+  def validate_each(record, attribute, value)
+    unless value.try(:length) == options[:is]
+      record.errors[attribute] << "is not of length #{options[:is]}."
+    end
+  end
```

Comment:
cool that you wrote your own, but this is a thing - http://guides.rubyonrails.org/active_record_validations.html#numericality